src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -326,6 +326,13 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		});
 		this._lifecycleService.onBeforeShutdown(e => {
 			this._willRestart = e.reason !== ShutdownReason.RELOAD;
+			this.getActiveTasks().then(async (tasks) => {
+				for (const task of tasks) {
+					if (!task.configurationProperties.isBackground) {
+						await this.terminate(task);
+					}
+				}
```

Comment:
Shutdown time is very tight, you'd be better off doing these all at once via `Promise.all` instead of in sequence